src/screens/staking-delegate-form.tsx

```diff
@@ -19,39 +20,42 @@ export const StakingDelegateFormScreen = () => {
   const wallet = Wallet.getById(account);
   const balances = useWalletsBalance([wallet!]);
   const currentBalance = useMemo(() => balances[account], [balances, account]);
-  const [fee, setFee] = useState<Balance | null>(null);
+  const [fee, _setFee] = useState<Balance | null>(null);
 
   const setDefaultFee = useCallback(
-    () => setFee(new Balance(Cosmos.fee.amount)),
+    () => _setFee(new Balance(Cosmos.fee.amount)),
     [],
   );
 
-  useLayoutEffectAsync(async () => {
-    const timer = setTimeout(() => {
-      setDefaultFee();
-    }, FEE_ESTIMATING_TIMEOUT_MS);
-
-    const instance = await getProviderInstanceForWallet(wallet!, true);
+  const setFee = useThrottle(async (amount?: string) => {
     const cosmos = new Cosmos(app.provider);
+    const instance = await getProviderInstanceForWallet(wallet!, true);
+
     try {
-      setFee(null);
+      _setFee(null);
       const f = await cosmos.simulateDelegate(
         instance,
         wallet!.path!,
         validator.operator_address,
-        currentBalance.availableForStake,
+        amount ? new Balance(amount) : currentBalance.availableForStake,
       );
       Logger.log('f.amount', f.amount);
-      setFee(new Balance(f.amount));
+      _setFee(new Balance(f.amount));
```

Comment:
In fee calculation we need to add something extra. For example metamask adds 20% of total value to the gas amount.

Reply:
Yes, f.amount already increased with 20% inside simulateDelegate function